### PR TITLE
[fix][test] Fix flaky test LeaderElectionTest.revalidateLeaderWithinSameSession

### LINKED ISSUE #21535: Flaky-test: LeaderElectionTest.revalidateLeaderWithinSameSession
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failures

- [2023-11-01T22:34:20.4026218Z](https://github.com/apache/pulsar/actions/runs/6725651280/job/18280799117#step:11:1934) 
- [2023-10-31T18:55:24.5179215Z](https://github.com/apache/pulsar/actions/runs/6710630191/job/18237677637#step:11:1899) 

### Exception stacktrace

```
java.lang.AssertionError: expected [Optional[test-2]] but found [Optional.empty]
	at org.testng.Assert.fail(Assert.java:110)
	at org.testng.Assert.failNotEquals(Assert.java:1577)
	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
	at org.testng.Assert.assertEquals(Assert.java:131)
	at org.testng.Assert.assertEquals(Assert.java:643)
	at org.apache.pulsar.metadata.BaseMetadataStoreTest.lambda$assertEqualsAndRetry$6(BaseMetadataStoreTest.java:133)
	at org.apache.pulsar.metadata.BaseMetadataStoreTest.retryStrategically(BaseMetadataStoreTest.java:141)
	at org.apache.pulsar.metadata.BaseMetadataStoreTest.assertEqualsAndRetry(BaseMetadataStoreTest.java:129)
	at org.apache.pulsar.metadata.BaseMetadataStoreTest.assertEqualsAndRetry(BaseMetadataStoreTest.java:121)
	at org.apache.pulsar.metadata.LeaderElectionTest.revalidateLeaderWithinSameSession(LeaderElectionTest.java:212)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by nikam14**
Fixes #21535 

### Motivation
The test should Provide Consistent Answer .Test revalidateLeaderWithinSameSession failed too many times. 

### Modifications
 We are calling the method once store the object and check both the conditions . In this way at both conditions we are checking the same Object.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BaseMetadataStoreTest.java [hunk 1]
```diff
@@ -145,10 +145,11 @@ public static void assertEqualsAndRetry(Supplier<Object> actual,
                                             int retryCount,
                                             long intSleepTimeInMillis) throws Exception {
         assertTrue(retryStrategically((__) -> {
-            if (actual.get().equals(expectedAndRetry)) {
+            Object Actual = actual.get();
```
Comment (hanmz) [inline pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BaseMetadataStoreTest.java:148]: It is recommended that local variables start with a lowercase letter

**Comment by nikam14** [comment]
@lhotari  could you please review the pr

**Review by lhotari — APPROVED** [review]

**Review by BewareMyPower — APPROVED** [review]

**Comment by BewareMyPower** [comment]
This PR is 544 commits behind the master. Could you merge the master branch?

**Comment by lhotari** [comment]
> This PR is 544 commits behind the master. Could you merge the master branch?

@BewareMyPower I'll close and reopen this PR to trigger a new PR build. In Pulsar CI, the PR build will always merge the master branch before running the PR build.

**Pull request merged**